File: bindings/python/examples/provider_chat.py
from __future__ import annotations

from cogentlm import ChatMessage

from _common import (
    load_openai_provider_client,
    print_text,
    provider_endpoint,
    read_provider_args,
    text_options,
)


def main() -> None:
    model, prompt = read_provider_args("Explain provider inference in one sentence.")
    client = load_openai_provider_client(model)
    run = client.chat(
        [
            ChatMessage("system", "Answer concisely."),
            ChatMessage("user", prompt),
        ],
        endpoint=provider_endpoint(model),
        options=text_options(),
        stream_tokens=True,
    )
    streamed = ""
    for batch in run.tokens():
        print(batch["text"], end="", flush=True)
        streamed += batch["text"]
    print()
    result = run.result()
    if streamed != result["text"]:
        raise RuntimeError("streamed token batches did not match final response text")
    print_text(result)


if __name__ == "__main__":
    main()

File: bindings/node/examples/provider_chat.mjs
import {
  loadOpenAiProviderClient,
  printText,
  providerEndpoint,
  readProviderArgs,
  textOptions,
} from './_common.mjs';

const { model, input } = readProviderArgs('Explain provider inference in one sentence.');
const client = loadOpenAiProviderClient(model);
const run = client.chat({
  endpoint: providerEndpoint(model),
  messages: [
    { role: 'system', content: 'Answer concisely.' },
    { role: 'user', content: input },
  ],
  options: textOptions(),
  streamTokens: true,
});
let streamed = '';
for await (const batch of run.tokens) {
  process.stdout.write(batch.text);
  streamed += batch.text;
}
process.stdout.write('\n');
const result = await run.response;
if (streamed !== result.text) {
  throw new Error('streamed token batches did not match final response text');
}
printText(result);
